### DISCUSSION TITLE
Handshaking in srt

### DISCUSSION THREAD

**Discussion by youssef-haidar** [Haivision/srt#2786] [Q&A]
I was wondering that what would happen to the connection when at some points the link fails for some reasons. Is it going to do the handshaking again? In other words the handshaking happens just at the beginning or it can happen again later?
If yes can somebody give examples!

Thanks in advance

**Comment by ethouris** [comment]
This works exactly the same as with TCP.

**Reply by youssef-haidar** [reply]
So you mean it will re-establish the connection by performing a new handshaking

**Reply by youssef-haidar** [reply]
It is still not clear for me

**Reply by maxsharabayko** [reply]
The connection will get broken and closed. The application can try to reconnect if it wants to.

**Reply by Satellite-Bruce** [reply]
That is the rub isn't it. I see in my logging a successful call/handshake and at some point, from packet loss etc, a disconnect. I have to believe that for the connection to be reestablished a new handshake must occur. Trying to figure out how to get my Haivision decoder (the caller) to be more persistent. If a connection does not exist make a new call. Seems simple and so far not. I'll post
back if I resolve it.
